### DISCUSSION TITLE
predict_dataloader is missing in LightningDataModule

### DISCUSSION THREAD

**Discussion by digital-idiot** [Lightning-AI/pytorch-lightning#12893] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
In the [documentation](https://pytorch-lightning.readthedocs.io/en/stable/extensions/datamodules.html#predict-dataloader) it is said that:
> Use the [predict_dataloader()](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.core.hooks.DataHooks.html#pytorch_lightning.core.hooks.DataHooks.predict_dataloader) method to generate the prediction dataloader(s)

However, unlike `train_dataloader`, `val_dataloader` and `test_dataloader` there is no reference of `predict_dataloader` in the [source](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/core/datamodule.py). Also `from_dataset` class method does not have any argument for passing `predict_dataset`.

**Comment by rohitgr7** [comment]
hey @digital-idiot !
good catch! would you like to send a PR to add it?

**Comment by digital-idiot** [comment]
@rohitgr7 
Yes, I may, but it may take a day or two.

**Comment by digital-idiot** [comment]
@rohitgr7 
When I was trying use `LightningDataModule` I also noticed that [`train_dataloader`](https://github.com/PyTorchLightning/pytorch-lightning/blob/eebba9e632a53a2bc945ddec430aec45f315b9d1/pytorch_lightning/core/datamodule.py#L120) is using `shuffle=True` by default which user can't change also for my personal project I needed to provide `collate_fn` which I can't do ATM without overloading the method. Would it be acceptable if the API just provides a general `**kwargs` option where we can pass all the options supported by [DataLoader](https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader)?

**Reply by rohitgr7** [reply]
tbh, I was going to propose to deprecate and remove this method 😅  since it just solved 2 lines of code and doesn't let users to follow the general approach of using `prepare_data` and `setup`.

also adding `**kwargs` would make it difficult to configure them since we can't say whether they should be added for all dataloaders or just any one of them. So I'd always recommend using `setup` and `prepare_data` and override dataloader methods to define the dataloader.
